shopfloor/services/zone_picking.py

```diff
@@ -603,8 +623,17 @@ def _scan_source_package(
         move_lines = self._find_location_move_lines(
             locations=sublocation, package=package
         )
+
+        handle_complete_mix_pack = self._handle_complete_mix_pack(package)
+        handling_package = None
+        if handle_complete_mix_pack:
+            handling_package = package
+
```

Comment:
This should be in the below `if` statement. Handling this complete mix pack separately makes the code more readable. Then you don't have to change the code afterwards
```suggestion
        if move_lines:
            handle_complete_mix_pack = self._handle_complete_mix_pack(package)
            if handle_complete_mix_pack:
                move_line = first(move_lines)
                response = self._response_for_set_line_destination(
                    move_line, qty_done=0, package=package
                )
            return response, message
```